src/Acuminator/Acuminator.Utilities/DiagnosticSuppression/SuppressionFile.cs

```diff
@@ -78,17 +76,19 @@ internal static bool IsSuppressionFile(string path)
 				messages = LoadMessages(fileSystemService, suppressionFilePath);
 			}
 
-			var fileWatcher = fileSystemService.CreateWatcher(suppressionFilePath);
+			ISuppressionFileWatcherService fileWatcher;
 
+			lock (fileSystemService)
```

Comment:
You already have a lock on `fileSystemService` in the call stack above the current method

Reply:
The lock allows recursive calls by the same thread and at the same time we cannot guarantee that only `SuppressionManager` will call the `SuppressionFile`. It's a public class which accepts some file service so I thought that it's better to make sure that the service is accessed exclusively

Reply:
@SENya1990 why do we need a look here?

Reply:
@metallmonster because the VS throws exception about access to file from another thread.

Reply:
@metallmonster because the VS throws exception about access to file from another thread.